lib/src/builder/parsing/declarations_from_members.dart

```diff
@@ -209,6 +210,12 @@ Iterable<BoilerplateDeclaration> getBoilerplateDeclarations(
     ..sort((a, b) => b.versionConfidences.maxConfidence.confidence
         .compareTo(a.versionConfidences.maxConfidence.confidence));
   for (final factory in factoriesMostToLeastConfidence) {
+    if(factory.versionConfidences.maxConfidence.version == Version.v5_functionComponent) {
```

Comment:
Do we need a v5? We already have logic in place that yields a `PropsMapViewOrFunctionComponentDeclaration`; could we just use that and build in some factory grouping logic beforehand?

Reply:
That's true, I didn't think about how I could just check if `uiFunctionComponent` was called here instead of when the members are detected. That would make it a lot simpler!

And I made a `FunctionComponentDeclaration` class separately from `PropsMapViewOrFunctionComponentDeclaration` because I wanted `factory` to be a list of factories, but I guess I could just change the props map view logic to look for a single factory in the list.

Thanks for the feedback!

Reply:
Sweet, sounds good. Np!